src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,135 @@
+     * Here are requirements for parameters:
+     * max_token_count and token_limit should be a positive integer
+     * overlap_rate should be within range [0, 0.5]
+     * tokenizer should be string
+     */
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        this.tokenLimit = validatePositiveIntegerParameter(parameters, TOKEN_LIMIT_FIELD, DEFAULT_TOKEN_LIMIT);
+        if (parameters.containsKey(OVERLAP_RATE_FIELD)) {
+            String overlapRateString = parameters.get(OVERLAP_RATE_FIELD).toString();
+            if (!(NumberUtils.isParsable(overlapRateString))) {
+                throw new IllegalArgumentException(
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] cannot be cast to [" + Number.class.getName() + "]"
+                );
+            }
+            double overlapRate = NumberUtils.createDouble(overlapRateString);
+            if (overlapRate < 0 || overlapRate > OVERLAP_RATE_UPPER_BOUND) {
+                throw new IllegalArgumentException(
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] must be between 0 and " + OVERLAP_RATE_UPPER_BOUND
+                );
+            }
+            this.overlapRate = overlapRate;
+        } else {
+            this.overlapRate = DEFAULT_OVERLAP_RATE;
+        }
+        this.tokenizer = validateStringParameters(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER, false);
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param runtimeParameters a map for runtime parameters, containing the following runtime parameters:
+     * max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(String content, Map<String, Object> runtimeParameters) {
+        // prior to chunking, runtimeParameters have been validated
+        int maxTokenCount = validatePositiveIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
+
+        List<AnalyzeToken> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> passages = new ArrayList<>();
+
+        int startTokenIndex = 0;
+        int startContentPosition, endContentPosition;
+        int overlapTokenNumber = (int) Math.floor(tokenLimit * overlapRate);
+
+        while (startTokenIndex < tokens.size()) {
+            startContentPosition = tokens.get(startTokenIndex).getStartOffset();
+            if (startTokenIndex + tokenLimit >= tokens.size()) {
+                // include all characters till the end if no next passage
+                endContentPosition = content.length();
+                passages.add(content.substring(startContentPosition, endContentPosition));
+                break;
+            } else {
+                // include gap characters between two passages
+                endContentPosition = tokens.get(startTokenIndex + tokenLimit).getStartOffset() - 1;
```

Comment:
If we input "a   b", the output would be ["a ", "b"]. Is the whitespace after "a" what we want? Or do we need to change to `endContentPosition = tokens.get(startTokenIndex + tokenLimit - 1).getEndOffset()`

Reply:
BUG: for some tokenizers like `thai`, the start offset of next token could be end of previous. By -1, we will lose some tokens.

Reply:
> Is the whitespace after "a" what we want?

Yes, we do not wish to discard any characters from the raw content.

Reply:
> > Is the whitespace after "a" what we want?
> 
> Yes, we do not wish to discard any characters from the raw content.

But for something like "a b", the result would be "a" and "b", why we remove white space here?

Reply:
Anyway, we need to remove `-1` to make sure the `thai` tokenizer works well.

Reply:
Already changed.